examples/infection_phenotyping/Infection_classification_model.py

```diff
@@ -0,0 +1,251 @@
+
+# from monai.losses import DiceLoss
+# from viscy.light.engine import VSUNet
+from viscy.unet.networks.Unet2D import Unet2d
+from viscy.data.hcs import Sample
+from viscy.transforms import RandWeightedCropd, RandGaussianNoised
+from viscy.transforms import NormalizeSampled
+
+# %% Create a dataloader and visualize the batches.
+# Set the path to the dataset
+dataset_path = "/hpc/projects/intracellular_dashboard/viral-sensor/infection_classification/datasets/Exp_2023_09_27_DENV_A2_infMarked_refined.zarr"
+
+# Create an instance of HCSDataModule
+data_module = HCSDataModule(
+    dataset_path,
+    source_channel=["Sensor", "Phase"],
+    target_channel=["Inf_mask"],
+    yx_patch_size=[128, 128],
+    split_ratio=0.8,
+    z_window_size=1,
+    architecture="2D",
+    num_workers=1,
+    batch_size=64,
+    normalizations=[
+        NormalizeSampled(
+            keys=["Sensor", "Phase"],
+            level="fov_statistics",
+            subtrahend="median",
+            divisor="iqr",
+        )
+    ],
+    augmentations=[
+        RandWeightedCropd(
+            num_samples=8,
+            spatial_size=[-1, 128, 128],
+            keys=["Sensor", "Phase", "Inf_mask"],
+            w_key="Inf_mask",
+        ),
+        RandGaussianNoised(keys=["Sensor", "Phase"], mean=0.0, std=1.0, prob=0.5),
+    ],
+)
+
+# Prepare the data
+data_module.prepare_data()
+
+# Setup the data
+data_module.setup(stage="fit")
+
+# Create a dataloader
+train_dm = data_module.train_dataloader()
+
+val_dm = data_module.val_dataloader()
+
+# Visualize the dataset and the batch using napari
+# Set the display
+# os.environ['DISPLAY'] = ':1'
+
+# # Create a napari viewer
+# viewer = napari.Viewer()
+
+# # Add the dataset to the viewer
+# for batch in dataloader:
+#     if isinstance(batch, dict):
+#         for k, v in batch.items():
+#             if isinstance(v, torch.Tensor):
+#                 viewer.add_image(v.cpu().numpy().astype(np.float32))
+
+# # Start the napari event loop
+# napari.run()
+
+# %% use 2D Unet and Lightning module
+
+
+# Train the model
+# Create a TensorBoard logger
+class LightningUNet(pl.LightningModule):
+    def __init__(
+        self,
+        in_channels,
+        out_channels,
+        lr: float = 1e-3,
+        loss_function: nn.CrossEntropyLoss = None,
+        schedule: Literal["WarmupCosine", "Constant"] = "Constant",
+        log_batches_per_epoch: int = 2,
+        log_samples_per_batch: int = 1,
+    ):
+        super(LightningUNet, self).__init__()
+        self.unet_model = Unet2d(in_channels=in_channels, out_channels=out_channels)
+        self.lr = lr
+        self.loss_function = loss_function if loss_function else nn.CrossEntropyLoss()
+        self.schedule = schedule
+        self.log_batches_per_epoch = log_batches_per_epoch
+        self.log_samples_per_batch = log_samples_per_batch
+        self.training_step_outputs = []
+        self.validation_step_outputs = []
+
```

Comment:
Adding suggestions to copute and plot confusion matrix.
```
        self.val_cm = torchmetrics.classification.ConfusionMatrix(
            task="multiclass", num_classes=self.n_classes
        )
```

Reply:
Above call creates an object that can accumulate the confusion matrix over an entire epoch.